File: Retrofit/app/src/main/java/com/example/retrofit/MainActivity.java
package com.example.retrofit;

import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.example.retrofit.databinding.ActivityMainBinding;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity implements MainInterface{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityMainBinding activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        Model model = new Model();
        MainViewModel mainViewModel = new MainViewModel(this, model);
        activityMainBinding.setMainViewModel(mainViewModel);
        activityMainBinding.setModel(model);
        activityMainBinding.getMainViewModel();
        activityMainBinding.recyclerView.setLayoutManager(new LinearLayoutManager(this));
        activityMainBinding.recyclerView.setHasFixedSize(true);


    }

    @Override
    public void executed() {
        Toast.makeText(MainActivity.this,"executed",Toast.LENGTH_SHORT).show();

    }
}

File: Multiple EditText Single TextWatcher/app/src/main/java/com/fidelity/multipleedittextsingletextwatcher/MainActivity.kt
package com.fidelity.multipleedittextsingletextwatcher

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        /// Create the view model
        val mainViewModel = MainViewModel()

        /// Data-Binding
        val binding = DataBindingUtil.setContentView<ViewDataBinding>(this, R.layout.activity_main)
        binding.setVariable(BR.mainViewModel, mainViewModel)

    }
}
